lib/wings/valkyrie/persister.rb

```diff
@@ -68,6 +71,42 @@ def initialize(msg = nil, obj:)
           super(msg)
         end
       end
+
+      private
+
+        ##
+        # @return [void]
+        # @raise [::Valkyrie::Persistence::StaleObjectError]
+        def check_lock_tokens(af_object:, resource:)
+          return if !resource.optimistic_locking_enabled? || af_object.new_record?
```

Comment:
I find myself preferring to read single step guards. Check my "math" on this, but are they equivalent?

Current:

```ruby
return if !resource.optimistic_locking_enabled? || af_object.new_record?
return if
            etag_lock_token_valid?(af_object: af_object, resource: resource) &&
            last_modified_lock_token_valid?(af_object: af_object, resource: resource)
```

Preferred:

```ruby
return unless resource.optimistic_locking_enabled?
return if af_object.new_record?
return unless etag_lock_token_valid?(af_object: af_object, resource: resource)
return unless last_modified_lock_token_valid?(af_object: af_object, resource: resource)
```

I find my preferred (assuming its correct) to read more quickly. I don't need to mentally juggle `&&` or `! ||`

Reply:
seems reasonable to me. i probably went through 5 refactorings of this to get it where it is. no harm in a few more :)

Reply:
re: your math, i think we still need to evaluate the two lock tokens together; we can't short-circuit until both pass.

the alternative is:
```ruby
raise [...] unless etag_lock_token_valid?(af_object: af_object, resource: resource)
raise [...] unless last_modified_lock_token_valid?(af_object: af_object, resource: resource)
```

but i rejected this due to the duplicated `raise` code.